### ISSUE TITLE
Lightning trap's behavior is bugged, attacks any unit within a 3-tile radius when it's duration is over

### ISSUE THREAD

**Issue by Whurrhurr** [Loobinex/keeperfx-unofficial#58]
At some point during 4.6 unofficial the behavior lighting traps had (shoot lightning at the closest unit to it when triggered) bugged out (unless this change was intentional. never saw any patch note saying that this trap's behavior was changed.) and now when the trap's duration is over it damages every unit within a (i think) 3 tile radius from the trap. even if the unit is from the same team as the trap's owner (i.e green keeper's units get hurt by the lightning trap from this bugged attack when the trap's duration is over even though the trap is owned by green.) this makes lightning traps very effective at weakening large groups of units or even outright killing dozens of units when it shouldn't be able to (went into hero heart room in blaise end by using destroy walls with 60 units. ended with only a mere 17 units. i lost 43 units to the 4 lightning traps on the hero heart tiles.) only got around to reporting it now since i didn't see any post/thread on keeperklan about it (may have missed a post about it) or an issue reported on the git.

**Comment by spartahawk** [comment]
@Whurrhurr You're right. I'm seeing it in the most recent version at least. And it looks like the damage occurs right at the end of the lightning trap, though no lightning hits the bystander. I'll look into it. Thanks.

**Comment by spartahawk** [comment]
There's definitely splash damage that decreases with distance.

I might be applying the patch wrong, but I'm even seeing this effect after patching with the official KeeperFX v0.4.6 r1834 patch for example. So I think it predates Unofficial.

@Loobinex were you aware of this already and do you know if it was intentional?

**Comment by spartahawk** [comment]
Okay apparently this area damage was added to several traps including lightning in 2013.
https://github.com/dkfans/keeperfx-stable/commit/a8be34d6e14f62bff1595963722075df4777fb11#diff-5300affb36a67638cfabd683d0eae044R453

Apparently area damage affects all creatures and I'm not sure if that is easily changeable. But if there is demand to make trap area damage not affect friendlies, we can probably do that.

I'd say just remove the area damage from lightning traps, or decrease it. I don't think it makes sense for lightning to have area damage, personally.

Lightning trap was changed to not directly hit allies at one point. https://github.com/dkfans/keeperfx-stable/commit/c308a93c87d32b74273aebd08244033bcb467235 but the area damage still affects all creatures.

**Comment by Loobinex** [comment]
I think this is worth looking into some more, but I don't think simply removing the area damage is the right approach. (The keeper lightning spell also does area damage when charged by the way)

Currently, the thing with traps is that they do very little against units, and are very powerful against workers. Traps doing area damage mitigates this somewhat, since that is mostly effective against groups of enemy fighters.

The thing is, whatever happens, the player should get proper feedback. I'll get back on this.
